pkg/flags/flags.go

```diff
@@ -27,20 +27,26 @@ import (
 
 // Options has fields for the existing flags
 type Options struct {
-	Action, Release, Channel, Arch, LogLevel, Qcow2compat, OS, Kernel, Gadget, ImageType string
+	Action, Release, Channel,
+	Arch, LogLevel, Qcow2compat,
+	OS, Kernel, Gadget, ImageType,
+	OSChannel, GadgetChannel, KernelChannel string
 }
 
 const (
-	defaultAction      = "create"
-	defaultRelease     = "rolling"
-	defaultChannel     = "edge"
-	defaultArch        = "amd64"
-	defaultLogLevel    = "info"
-	defaultQcow2compat = "1.1"
-	defaultKernel      = "canonical-pc-linux.canonical"
-	defaultOS          = "ubuntu-core.canonical"
-	defaultGadget      = "canonical-pc.canonical"
-	defaultImageType   = "custom"
+	defaultAction        = "create"
+	defaultRelease       = "rolling"
+	defaultChannel       = "edge"
```

Comment:
You are right, I've added the changes to remove it. I've also taken the opportunity to prevent having all the snaps sideloaded, now a base channel is calculated (the one more requested or the os channel if all of them are different) and only the snap from other channels are sideloaded.